vectordatasource/transform.py

```diff
@@ -3657,6 +3657,76 @@ def drop_features_mz_min_pixels(ctx):
             feature_layer['features'] = features_to_keep
 
 
+def _drop_small_inners(poly, area_tolerance):
+    ext = poly.exterior
+
+    inners = []
+    for inner in poly.interiors:
+        area = Polygon(inner).area
+        if area >= area_tolerance:
+            inners.append(inner)
+
+    return Polygon(ext, inners)
+
+
+def drop_small_inners(ctx):
+    """
+    Drop inners which are smaller than the given scale.
+    """
+
+    zoom = ctx.nominal_zoom
+    start_zoom = ctx.params.get('start_zoom', 0)
+    end_zoom = ctx.params.get('end_zoom')
+    pixel_area = ctx.params.get('pixel_area')
+    source_layers = ctx.params.get('source_layers')
+
+    assert source_layers, \
+        "You must provide source_layers (layer names) to drop_small_inners"
+    assert pixel_area, \
+        "You must provide a pixel_area parameter to drop_small_inners"
+
+    if zoom < start_zoom:
+        return None
+
+    if end_zoom and zoom > end_zoom:
+        return None
+
+    meters_per_pixel_area = calc_meters_per_pixel_area(zoom)
+    area_tolerance = meters_per_pixel_area**2 * pixel_area
```

Comment:
Well spotted!

Yes, I seem to have been confused about whether that was a linear or area ratio. D'oh!

Fixed in 71488a5.